gateway/ln-gateway/src/lnd.rs

```diff
@@ -0,0 +1,302 @@
+                        )))
+                        .await;
+                    None
+                }
+            } {
+                trace!("handling htlc {:?}", htlc);
+                let response: Option<ForwardHtlcInterceptResponse> =
+                    if htlc.outgoing_requested_chan_id != scid {
+                        // Pass through: This HTLC doesn't belong to the current subscription
+                        // Forward it to the next interceptor or next node
+                        Some(ForwardHtlcInterceptResponse {
+                            incoming_circuit_key: htlc.incoming_circuit_key,
+                            action: 2,
+                            preimage: vec![],
+                            failure_message: vec![],
+                            failure_code: 0,
+                        })
+                    } else {
+                        // TODO: generate unique id for each intercepted HTLC
+                        let intercepted_htlc_id = sha256::Hash::hash(&htlc.onion_blob);
+
+                        // Intercept: This HTLC belongs to the current subscription
+                        let intercept = SubscribeInterceptHtlcsResponse {
+                            payment_hash: htlc.payment_hash,
+                            incoming_amount_msat: htlc.incoming_amount_msat,
+                            outgoing_amount_msat: htlc.outgoing_amount_msat,
+                            incoming_expiry: htlc.incoming_expiry,
+                            short_channel_id: scid,
+                            intercepted_htlc_id: intercepted_htlc_id.into_inner().to_vec(),
+                        };
+
+                        // Send it to gatewayd for processing
+                        match gwd_tx.send(Ok(intercept)).await {
+                            Ok(_) => {
+                                // Keep a reference to LND sender reference so we can later forward
+                                // outcomes on `complete_htlc` rpc
+                                outcomes.lock().await.insert(
+                                    intercepted_htlc_id,
+                                    (Arc::new(lnd_tx.clone()), htlc.incoming_circuit_key),
+                                );
+
+                                None
+                            }
+                            Err(e) => {
+                                error!("Failed to send HTLC to gatewayd for processing: {:?}", e);
+                                Some(cancel_intercepted_htlc(htlc.incoming_circuit_key))
+                            }
+                        }
+                    };
+
+                if response.is_some() {
+                    // TODO: Consider retrying this if the send fails
+                    let _ = lnd_tx.send(response.unwrap()).await.map_err(|e| {
+                        error!("Failed to send response to LND: {:?}", e);
+                        // The HTLC will timeout and LND will automatically cancel it.
```

Comment:
This could be a long timeout though right?

Reply:
correct